

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Biomedical Application &mdash; Exascale Knowledge Share 0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=2709fde1"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Extreme-scale Scientific Software Stack (E4S)" href="e4s.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            Exascale Knowledge Share
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="home.html">Home</a></li>
<li class="toctree-l1"><a class="reference internal" href="quickStart.html">Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="alphafold.html">AlphaFold 3 :  The inference pipeline</a></li>
<li class="toctree-l1"><a class="reference internal" href="mathphysic.html">Mathematical modeling in Physics Engineering, Biological and Social Science with QwQ LLM from China</a></li>
<li class="toctree-l1"><a class="reference internal" href="rstudio.html">Mathematics and Statistics Tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="wine.html">Excel Macro Runner Docker</a></li>
<li class="toctree-l1"><a class="reference internal" href="container.html">Apptainer/Singularity</a></li>
<li class="toctree-l1"><a class="reference internal" href="workshop.html">Workshop</a></li>
<li class="toctree-l1"><a class="reference internal" href="swguide.html">Software Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="swguide.html#macromolecular-modeling-and-visualization">Macromolecular Modeling and Visualization:</a></li>
<li class="toctree-l1"><a class="reference internal" href="swguide.html#gromacs-high-performance-molecular-dynamics">GROMACS: High Performance Molecular Dynamics</a></li>
<li class="toctree-l1"><a class="reference internal" href="swguide.html#pilot-test-cluster-access-and-performance-awareness">Pilot Test: Cluster Access and Performance Awareness:</a></li>
<li class="toctree-l1"><a class="reference internal" href="powersaving.html">Electrical Energy  Saving</a></li>
<li class="toctree-l1"><a class="reference internal" href="quantumml.html">Quantum Machine Learning</a></li>
<li class="toctree-l1"><a class="reference internal" href="torch_multi.html">Pytorch DISTRIBUTED DATA PARALLEL</a></li>
<li class="toctree-l1"><a class="reference internal" href="parabricks.html">NVIDIA Parabricks</a></li>
<li class="toctree-l1"><a class="reference internal" href="cryo_em.html">Cryogenic electron microscopy</a></li>
<li class="toctree-l1"><a class="reference internal" href="cryo_em.html#relion-4-0-for-cryo-em-structure-determination">Relion 4.0 for cryo-EM structure determination</a></li>
<li class="toctree-l1"><a class="reference internal" href="cryo_em.html#computational-imaging-system-for-transmission-electron-microscopy-cistem">Computational Imaging System for Transmission Electron Microscopy(cisTEM)</a></li>
<li class="toctree-l1"><a class="reference internal" href="llm.html">LLaMA</a></li>
<li class="toctree-l1"><a class="reference internal" href="llm.html#lora-ai-at-the-edge-is-comming">LoRA: AI at the edge is comming.</a></li>
<li class="toctree-l1"><a class="reference internal" href="llm.html#goolge-s-llm-gemma">Goolge’s LLM: Gemma</a></li>
<li class="toctree-l1"><a class="reference internal" href="llm.html#responsible-ai">Responsible AI</a></li>
<li class="toctree-l1"><a class="reference internal" href="llm.html#get-source-repository-and-make-them-work-on-k8s-for-ollama">Get Source repository and make them work on K8S  for Ollama</a></li>
<li class="toctree-l1"><a class="reference internal" href="llm.html#restapi-test-on-host-that-port-forwarded">RestAPI test on host that port forwarded</a></li>
<li class="toctree-l1"><a class="reference internal" href="llm.html#multi-lingual-capabilities-in-llama3-1-405b">Multi-lingual capabilities in Llama3.1 405B</a></li>
<li class="toctree-l1"><a class="reference internal" href="llm.html#api-inferencing-check-available-model">API inferencing check  available model :</a></li>
<li class="toctree-l1"><a class="reference internal" href="e4s.html">Extreme-scale Scientific Software Stack (E4S)</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Biomedical Application</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#blast">BLAST</a></li>
<li class="toctree-l2"><a class="reference internal" href="#allocate-resource-for-testing">Allocate resource for testing</a></li>
<li class="toctree-l2"><a class="reference internal" href="#making-example-file">Making example file</a></li>
<li class="toctree-l2"><a class="reference internal" href="#blasting">Blasting</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="#reproducting-research-results-with-python-and-conda-package-management">Reproducting research results with Python and Conda Package management</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#how-to">How to</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="#nextflow-reproducible-scientific-workflows">Nextflow:Reproducible scientific workflows</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#run-nextflow-core-or-nf-core-alphafold2-workflow">Run Nextflow core or nf-core AlphaFold2 workflow</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Exascale Knowledge Share</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Biomedical Application</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/bioinfo.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="biomedical-application">
<h1>Biomedical Application<a class="headerlink" href="#biomedical-application" title="Link to this heading"></a></h1>
<p>The following is note taking for applications for Bioinformatics.</p>
<section id="blast">
<h2>BLAST<a class="headerlink" href="#blast" title="Link to this heading"></a></h2>
<p>BLAST (basic local alignment search tool)[3] is an algorithm and program for comparing primary biological sequence information, such as the amino-acid sequences of proteins or the nucleotides of DNA and/or RNA sequences.</p>
<p><a class="reference external" href="https://en.wikipedia.org/wiki/BLAST_(biotechnology)">BLAST</a></p>
</section>
<section id="allocate-resource-for-testing">
<h2>Allocate resource for testing<a class="headerlink" href="#allocate-resource-for-testing" title="Link to this heading"></a></h2>
<p>To test an example on a compute node</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>salloc<span class="w"> </span>-t<span class="w"> </span><span class="m">1</span>:0:0<span class="w"> </span>-c<span class="w"> </span><span class="m">32</span><span class="w"> </span>--mem<span class="o">=</span>64GB
<span class="gp">$ </span>ssh<span class="w"> </span>&lt;allocated<span class="w"> </span>node&gt;
</pre></div>
</div>
</section>
<section id="making-example-file">
<h2>Making example file<a class="headerlink" href="#making-example-file" title="Link to this heading"></a></h2>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span><span class="nb">printf</span><span class="w"> </span><span class="s2">&quot;&gt;NR_024570.1 Escherichia coli strain U 5/41 16S ribosomal RNA, partial sequence</span>
<span class="go">AGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAGCAGCTTGCTGCTTTGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGCACAAAGAGGGGGACCTTAGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTAGTAGGTGGGGTAACGGCTCACCTAGGCGACGATCCCTAGCTGGTCTGAGAGGATGA</span>
<span class="go">&gt;NR_169460.1 Pseudomonas aylmerensis strain S1E40 16S ribosomal RNA, partial sequence</span>
<span class="go">CTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAGCGGTAGAGAGAAGCTTGCTTCTCTTGAGAGCGGCGGACGGGTGAGTAATGCCTAGGAATCTGCCTGGTAGTGGGGGATAACGTTCGGAAACGGACGCTAATACCGCATACGTCCTACGGGAGAAAGCAGGGGACCTTCGGGCCTTGCGCTATCAGATGAGCCTAGGTCGGATTAGCTAGTTGGTGGGGTAATGGCTCACCAAGGCGACGATCCGTAACTGGTCTGAGAGGATGATCAGTCACACTGGAACTGA</span>
<span class="go">&quot; &gt; refs.fa</span>



<span class="gp">$ </span><span class="nb">printf</span><span class="w"> </span><span class="s2">&quot;&gt;Q1</span>
<span class="go">AGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAGCAGCTTGCGGGGGTGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGCACAAAGAGGGGGACCTTAGGGCCTCTTGCCCCCCCATGTGCCCAGATGGGATTAGCTAGTAGGTGGGGTAACGGCTCACCTAGGCGACGATCCCTAGCTGGTCTGAGAGGATGA</span>
<span class="go">&gt;Q2</span>
<span class="go">CTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAGCGGTAGAGAGAAGCTTGCTTCTCTTGAGAGCGGCGGACCCCCGAGTAATGCCTAGGAATCTGCCTGGTAGTGGGGGATAACGTTCGGAAACGGACGCTAATACCGCATACGTCCTACGGGAGAAAGCAGGGGACCTTCGGGCCTTGCGCTATCAGATGAGGGGGGGTCGGATTAGCTAGTTGGTGGGGTAATGGCTCACCAAGGCGACGATCCGTAACTGGTCTGAGAGGATGATCAGTCACACTGGAACTGA</span>
<span class="go">&gt;Q3_exact</span>
<span class="go">AGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGGTAACAGGAAGCAGCTTGCTGCTTTGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGCACAAAGAGGGGGACCTTAGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTAGTAGGTGGGGTAACGGCTCACCTAGGCGACGATCCCTAGCTGGTCTGAGAGGATGA</span>
<span class="go">&gt;Q4_exact</span>
<span class="go">CTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAGCGGTAGAGAGAAGCTTGCTTCTCTTGAGAGCGGCGGACGGGTGAGTAATGCCTAGGAATCTGCCTGGTAGTGGGGGATAACGTTCGGAAACGGACGCTAATACCGCATACGTCCTACGGGAGAAAGCAGGGGACCTTCGGGCCTTGCGCTATCAGATGAGCCTAGGTCGGATTAGCTAGTTGGTGGGGTAATGGCTCACCAAGGCGACGATCCGTAACTGGTCTGAGAGGATGATCAGTCACACTGGAACTGA</span>
<span class="go">&quot; &gt; queries.fa</span>
</pre></div>
</div>
</section>
<section id="blasting">
<h2>Blasting<a class="headerlink" href="#blasting" title="Link to this heading"></a></h2>
<p>To run blast tools in singularity container with specific command blastn</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>singularity<span class="w"> </span><span class="nb">exec</span><span class="w"> </span>/app/blast.2.11.sif<span class="w"> </span>blastn<span class="w"> </span>-query<span class="w"> </span>queries.fa<span class="w"> </span>-subject<span class="w"> </span>refs.fa<span class="w"> </span>-outfmt<span class="w"> </span><span class="s2">&quot;6 qseqid qlen sseqid slen length pident evalue bitscore&quot;</span><span class="w"> </span>-max_hsps<span class="w"> </span><span class="m">1</span><span class="w"> </span>-max_target_seqs<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sort<span class="w"> </span>-nrk<span class="w"> </span><span class="m">8</span><span class="w"> </span>&gt;<span class="w"> </span>blast-results.tmp
</pre></div>
</div>
<p>Typically we want things in table form, here’s one way to run it for that and sort by bitscore: Adding a header</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>cat<span class="w"> </span>&lt;<span class="o">(</span><span class="nb">printf</span><span class="w"> </span><span class="s2">&quot;qseqid\tqlen\tsseqid\tslen\tlength\tpident\tevalue\tbitscore\n&quot;</span><span class="o">)</span><span class="w"> </span>blast-results.tmp<span class="w"> </span>&gt;<span class="w"> </span>blast-results.tsv


<span class="gp">$ </span>rm<span class="w"> </span>blast-results.tmp
<span class="gp">$ </span>column<span class="w"> </span>-ts<span class="w"> </span><span class="s1">$&#39;\t&#39;</span><span class="w"> </span>blast-results.tsv
</pre></div>
</div>
</section>
</section>
<section id="reproducting-research-results-with-python-and-conda-package-management">
<h1>Reproducting research results with Python and Conda Package management<a class="headerlink" href="#reproducting-research-results-with-python-and-conda-package-management" title="Link to this heading"></a></h1>
<p>In many cases, researcher need system adminstrators to install new library for their project. This is somehow taking longer time then we expected. So one solution in sharing system cluster, we are learing how to use ‘module’ system environment or singularity container. They are general usage. For Python eco-system, conda package management and conflict library resolver save us a lot when it comes to customization library for us.</p>
<p>We use newly release(two years)  and request from user for PSSMPRO.</p>
<section id="how-to">
<h2>How to<a class="headerlink" href="#how-to" title="Link to this heading"></a></h2>
<p>Setup Conda (once time setup)</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>module<span class="w"> </span>load<span class="w"> </span>anaconda3

<span class="gp">$ </span>conda<span class="w"> </span>init<span class="w"> </span>bash
</pre></div>
</div>
<p>Then logout and relogin again.</p>
<p>Create Conda Environment(one time  only)</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>conda<span class="w"> </span>create<span class="w"> </span>-n<span class="w"> </span>bio<span class="w"> </span><span class="nv">python</span><span class="o">=</span><span class="m">3</span>.9<span class="w"> </span>scikit-learn<span class="w"> </span>pandas<span class="w"> </span>jupyter<span class="w"> </span>blast<span class="w"> </span>bioconductor-kebabs<span class="o">=</span><span class="m">1</span>.24.0<span class="w"> </span>-c<span class="w"> </span>conda-forge<span class="w"> </span>-c<span class="w"> </span>bioconda
</pre></div>
</div>
<p>Wait….!</p>
<p>Activate environment when you want to work on project environment</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">  $ </span>conda<span class="w"> </span>env<span class="w"> </span>list

<span class="gp">  $  </span>conda<span class="w"> </span>activate<span class="w"> </span>bio


<span class="go">Add new library to Working Environment</span>
</pre></div>
</div>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>pip<span class="w"> </span>install<span class="w"> </span>pssmpro

<span class="gp">$ </span>jupyter<span class="w"> </span>notebook
</pre></div>
</div>
<p>In side your notebook your can verify that you can work with new installed package</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">from pssmpro.features import create_pssm_profile</span>
</pre></div>
</div>
</section>
</section>
<section id="nextflow-reproducible-scientific-workflows">
<h1>Nextflow:Reproducible scientific workflows<a class="headerlink" href="#nextflow-reproducible-scientific-workflows" title="Link to this heading"></a></h1>
<p>Scientific workflow engines,nextflow are particularly useful for data-intensive domains including  bioinformatics and radioastronomy, where data analysis and processing is made up of a number of tasks to be repeatedly executed across large datasets.
The following guide is example the of combination of container and workflow engines can be very effective in enforcing reproducible, portable, scalable science.
Run a workflow using Singularity and Nextflow
———————————————</p>
<p>Load repository :</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">~/nextflow/test$ git clone https://github.com/nextflow-io/rnaseq-nf.git</span>

<span class="gp">$ </span><span class="nb">cd</span><span class="w"> </span>/home/snit.san/nextflow/test/rnaseq-nf


<span class="gp">$ </span>module<span class="w"> </span>load<span class="w"> </span>slurm

<span class="gp">$ </span>salloca<span class="w"> </span>–t<span class="w"> </span><span class="m">1</span>:0:0<span class="w"> </span>–gres<span class="o">=</span>gpu:1

<span class="gp">$ </span>ssh<span class="w"> </span>&lt;compute_node&gt;

<span class="gp">$ </span>module<span class="w"> </span>load<span class="w"> </span>singularity

<span class="gp">$ </span>module<span class="w"> </span>load<span class="w"> </span>nextflow

<span class="gp">$ </span>nextflow<span class="w"> </span>run<span class="w"> </span>main.nf<span class="w"> </span>-profile<span class="w"> </span>singularity
</pre></div>
</div>
<section id="run-nextflow-core-or-nf-core-alphafold2-workflow">
<h2>Run Nextflow core or nf-core AlphaFold2 workflow<a class="headerlink" href="#run-nextflow-core-or-nf-core-alphafold2-workflow" title="Link to this heading"></a></h2>
<p>What ever AlphaFold3 will change the way research working on protein folding and challenging AutoDock, how to run AlphaFold2 with workflow is noted as follow.</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>module<span class="w"> </span>load<span class="w"> </span>slurm

<span class="gp">$ </span>salloca<span class="w"> </span>–t<span class="w"> </span><span class="m">1</span>:0:0<span class="w"> </span>–gres<span class="o">=</span>gpu:1

<span class="gp">$ </span>ssh<span class="w"> </span>&lt;compute_node&gt;

<span class="gp">$ </span>module<span class="w"> </span>load<span class="w"> </span>singularity

<span class="gp">$ </span>module<span class="w"> </span>load<span class="w"> </span>nextflow
</pre></div>
</div>
<p>Load repository :</p>
<blockquote>
<div><p>$ cd ~/nextflow/</p>
<p>$ git clone <a class="reference external" href="https://github.com/nf-core/proteinfold.git">https://github.com/nf-core/proteinfold.git</a></p>
<p>$ cd rproteinfold</p>
<p>$ unset https_proxy</p>
<p>$ nextflow run nf-core/proteinfold –input ./assets/samplesheet.csv –outdir ./output –mode alphafold2 base –full_dbs false –alphafold2_model_preset monomer –use_gpu true     -profile singularity</p>
</div></blockquote>
<dl>
<dt>Reference:</dt><dd><p><a class="reference external" href="https://hackmd.io/&#64;AstrobioMike/command-line-blast-example">Command-line blast example</a></p>
<p><a class="reference external" href="https://github.com/deeprob/pssmpro">Generate PSSM profiles for protein sequences</a></p>
<p><a class="reference external" href="https://github.com/deeprob/ThioesteraseEnzymeSpecificity">Thioesterases based on ensemble learning</a>
<a class="reference external" href="https://pawseysc.github.io/singularity-containers/41-workflow-engines/index.html">Containers on HPC and Cloud with Singularity</a></p>
</dd>
</dl>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="e4s.html" class="btn btn-neutral float-left" title="Extreme-scale Scientific Software Stack (E4S)" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, MAI.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>